### DISCUSSION TITLE
Using env secrets in github actions

### DISCUSSION THREAD

**Discussion by stormsson** [community/community#32561] [Actions]
I can't find a way to use environment secrets inside actions.

the only seemingly accepted solution is from [stackoverflow](https://stackoverflow.com/questions/67964110/how-to-access-secrets-when-using-flutter-web-with-github-actions/67998780#67998780) that it basically say "it doesn't work , use this instead"

Is this the expected behavior?

---

I have these env variables defined:

<img width="394" alt="image" src="https://user-images.githubusercontent.com/1133032/189485774-c3e8e9cd-b724-4dd7-ad5d-14a964d0b46c.png">

this inside the action
```
env:
  ALCHEMY_BASEURL: ${{secrets.ALCHEMY_BASEURL}}
  ...
```

but this does not resolve when the workflow is running

```
##[debug]Evaluating job-level environment variables
##[debug]Evaluating: secrets.ALCHEMY_BASEURL
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'ALCHEMY_BASEURL'
##[debug]=> null
##[debug]Result: null
```

**Comment by levs42** [comment]
Posting this to whoever tries to resolve the issue while searching for "Evaluating String ... Result: null" from the debug log.

I was calling workflow B from the workflow A. Workflow B cannot access secrets of the workflow A. I have to add all required secrets to secrets list:
```yml
jobs:
  call-B-workflow:
    uses: ./.github/workflows/B-workflow.yml
    secrets:
      MY_SECRET: ${{ secrets.MY_SECRET }}
```
